src/components/GraphHeartbeats.js

```diff
@@ -0,0 +1,123 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Paper } from 'material-ui'
+import Divider from 'material-ui/Divider'
+import { BarChart, Bar, Cell } from 'recharts'
+import Theme from '../../style/theme'
+
+class GraphHeartbeats extends React.PureComponent {
+  static getColor(amt) {
+    if (amt > 1) {
+      return '#00E200'
+    }
+    if (amt === 0) {
+      return '#707070'
+    }
+    let r = Math.floor(((((1 - amt) / 0.5) * 228) + 228) / 2)
+    let g = Math.floor((amt / 0.5) * 228)
+    const b = 0
+    if (amt > 0.5) {
+      g = 228
+    } else {
+      r = 228
+    }
+    return `#${r.toString(16).padStart(2, '0')}${g.toString(16).padStart(2, '0')}${b.toString(16).padStart(2, '0')}`
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      activeIndex: 0,
+      summaryData: {}
+    }
+    this.processData()
+  }
+
+  processData() {
+    const numberOfDaysToView = 90
+
+    const today = new Date()
+    const startToday = new Date(today.getFullYear(), today.getMonth(), today.getDate(), 0, 0, 0, 0)
+    const { summaryData } = this.state
+
+    for (let i = 0; i < numberOfDaysToView; i += 1) {
+      const day = new Date(startToday.getTime() - (86400000 * i))
+      const dateString = day.toLocaleDateString()
+      summaryData[dateString] = {
+        count: 0, dateString, interval: 1000, expected: 86400, height: 1
+      }
+    }
+
+    Object.values(this.props.data).forEach((heartbeat) => {
```

Comment:
Use lodash - https://lodash.com/docs/4.17.5#values. It's generally more performant, looks a lot cleaner, and we should stay consistent since it's being across the app. It's a library that can be used for most array and object related stuff. For basic stuff (forEach, map, filter, reduce), it's fine if you stick with the inbuilt stuff.

Reply:
I'm under the impression this ends up just being a visual change of the code, but it was no trouble to change.